apps/coa/behaviours/send-email-notification.js

```diff
@@ -0,0 +1,144 @@
+const config = require('../../../config');
+const utilities = require('../../../lib/utilities');
+const _ = require('lodash');
+const NotifyClient = utilities.NotifyClient;
+const userConfirmationTemplateId = config.govukNotify.userConfirmationTemplateId;
+const caseworkerConfirmationTemplateId = config.govukNotify.caseworkerConfirmationTemplateId;
+const caseworkerEmail = config.govukNotify.caseworkerEmail;
+const notifyKey = config.govukNotify.notifyApiKey;
+const translation = require('../translations/src/en/fields.json');
+const notifyClient = new NotifyClient(notifyKey);
+
+module.exports = class SendEmailConfirmation {
+  
+  async sendUserEmailNotification(req) {
+    const personalisations = this.notifyPersonalisations;
+
+    // Retrieve the value from the session model
+    const details_updating_value = req.sessionModel.get('which-details-updating');
+    
+    // Retrieve the corresponding label from the translations file
+    const details_updating_label = translation['which-details-updating'].options[details_updating_value].label;
+
+    try {
+      if (notifyKey === 'USE_MOCK') {
+        req.log('warn', '*** Notify API Key set to USE_MOCK. Ensure disabled in production! ***');
+      }
+
+      await notifyClient.sendEmail(userConfirmationTemplateId, req.sessionModel.get('email'), {
+        personalisation: Object.assign({}, personalisations, {
+        reference_number: req.sessionModel.get('uniqueRefNumber'),
+        applying_for_whom: req.sessionModel.get('isApplicant') ? 'your' : req.sessionModel.get('nameWithPossession'),
+        applicant_full_name: req.sessionModel.get('applicant-full-name'),
+        applicant_dob: req.sessionModel.get('applicant-dob'),
+        nationality: req.sessionModel.get('applicant-nationality'),
+        has_applicant_unique_number: req.sessionModel.get('applicant-unique-number').length > 0 ? 'yes' : 'no',
+        applicant_unique_number: req.sessionModel.get('applicant-unique-number'),
+        has_passport_number: req.sessionModel.get('passport-number-details') ? 'yes' : 'no',
+        passport_number: req.sessionModel.get('passport-number-details'),
+        has_brp_number: req.sessionModel.get('brp-details') ? 'yes' : 'no',
+        brp_number: req.sessionModel.get('brp-details'),
+        has_arc_number: req.sessionModel.get('arc-details') ? 'yes' : 'no',
+        arc_number: req.sessionModel.get('arc-details'),
+        who_are_you: req.sessionModel.get('who-are-you'),
+        is_legal_rep: req.sessionModel.get('legal-representative-name') ? 'yes' : 'no',
+        leg_rep_full_name: req.sessionModel.get('legal-representative-name'),
+        leg_rep_email_address: req.sessionModel.get('email'),
+        has_rep_tel_number: req.sessionModel.get('telephone').length > 0 ? 'yes' : 'no',
+        leg_rep_tel_number: req.sessionModel.get('telephone'),
+        has_leg_details: req.sessionModel.get('steps').includes('/legal-details') && req.sessionModel.get('oisc-sra-number') ? 'yes' : 'no',
```

Comment:
This condition can only check the steps for `/legal-details. If the steps not included the field value will not be there. so `oisc-sra-number` can be removed.
req.sessionModel.get('steps').includes('/legal-details') ? 'yes' : 'no',